test/e2e/shoot/create_rotate-ca_delete.go

```diff
@@ -42,44 +43,62 @@ var _ = Describe("Shoot Tests", Label("Shoot"), func() {
 		ctx, cancel = context.WithTimeout(parentCtx, 10*time.Minute)
 		defer cancel()
 
-		patch := client.MergeFrom(f.Shoot.DeepCopy())
-		metav1.SetMetaDataAnnotation(&f.Shoot.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationRotateCAStart)
-		Expect(f.GardenClient.Client().Patch(ctx, f.Shoot, patch)).To(Succeed())
+		Eventually(func() error {
+			patch := client.MergeFrom(f.Shoot.DeepCopy())
+			metav1.SetMetaDataAnnotation(&f.Shoot.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationRotateCAStart)
+			return f.GardenClient.Client().Patch(ctx, f.Shoot, patch)
```

Comment:
If the request fails and is retried, the next `patch` will be an empty patch.
Probably, you just need to wrap the call itself in `Eventually`.